interfaces/apparmor/backend_test.go

```diff
@@ -1515,3 +1515,148 @@ func (s *backendSuite) TestDowngradeConfinement(c *C) {
 		c.Check(apparmor.DowngradeConfinement(), Equals, tc.expected, Commentf("unexpected result for %+v", tc))
 	}
 }
+
+func (s *backendSuite) TestPtraceTraceRule(c *C) {
+	restoreTemplate := apparmor.MockTemplate("template\n###SNIPPETS###\n")
+	defer restoreTemplate()
+	restore := release.MockAppArmorLevel(release.FullAppArmor)
+	defer restore()
+	restore = apparmor.MockIsHomeUsingNFS(func() (bool, error) { return false, nil })
+	defer restore()
+
+	needle := `deny ptrace (trace),`
+	for _, tc := range []struct {
+		opts     interfaces.ConfinementOptions
+		uses     bool
+		suppress bool
+		expected bool
+	}{
+		// strict, only suppress if suppress == true and uses == false
+		{
+			opts:     interfaces.ConfinementOptions{},
+			uses:     false,
+			suppress: false,
+			expected: false,
+		},
+		{
+			opts:     interfaces.ConfinementOptions{},
+			uses:     false,
+			suppress: true,
+			expected: true,
+		},
+		{
+			opts:     interfaces.ConfinementOptions{},
+			uses:     true,
+			suppress: false,
+			expected: false,
+		},
+		{
+			opts:     interfaces.ConfinementOptions{},
+			uses:     true,
+			suppress: true,
+			expected: false,
+		},
+		// devmode, only suppress if suppress == true and uses == false
+		{
+			opts:     interfaces.ConfinementOptions{DevMode: true},
+			uses:     false,
+			suppress: false,
+			expected: false,
+		},
+		{
+			opts:     interfaces.ConfinementOptions{DevMode: true},
+			uses:     false,
+			suppress: true,
+			expected: true,
+		},
+		{
+			opts:     interfaces.ConfinementOptions{DevMode: true},
+			uses:     true,
+			suppress: false,
+			expected: false,
+		},
+		{
+			opts:     interfaces.ConfinementOptions{DevMode: true},
+			uses:     true,
+			suppress: true,
+			expected: false,
+		},
+		// classic, never suppress
+		{
+			opts:     interfaces.ConfinementOptions{Classic: true},
+			uses:     false,
+			suppress: false,
+			expected: false,
+		},
+		{
+			opts:     interfaces.ConfinementOptions{Classic: true},
+			uses:     false,
+			suppress: true,
+			expected: false,
+		},
+		{
+			opts:     interfaces.ConfinementOptions{Classic: true},
+			uses:     true,
+			suppress: false,
+			expected: false,
+		},
+		{
+			opts:     interfaces.ConfinementOptions{Classic: true},
+			uses:     true,
+			suppress: true,
+			expected: false,
+		},
+		// classic with jail, never suppress
```

Comment:
Hmm, wasn't classic with jail supposed to be the same as strict?

Reply:
Done